java/ql/src/semmle/code/java/Member.qll

```diff
@@ -115,13 +116,14 @@ class Callable extends StmtParent, Member, @callable {
 
   /**
    * Holds if this callable may call the specified callable,
-   * taking overriding into account.
+   * taking virtual dispatch into account.
    */
   predicate polyCalls(Callable m) {
     this.calls(m)
     or
-    exists(Method mSuper, VirtualMethodAccess c | c.getCaller() = this and c.getMethod() = mSuper |
-      m.(Method).overrides(mSuper)
+    exists(Call call |
```

Comment:
Does this make the first disjunct of `this.calls(m)` redundant? Either way, would it be better to drop it?

Reply:
`this.calls(m)` gets you the static call target whereas the virtual dispatch resolution gets you a possible dynamic target.  There are two cases where a static target isn't also a possible dynamic target:
* A method declared on an interface or an abstract method.
* A method on an instantiation of a generic class.  The static target will be the instantiated version, whereas the dynamic target will be the source declaration.

I could see dropping the static targets, but it depends on the use cases for `polyCalls`.

Reply:
I see. It'd be nice to explain that in doc, and possibly add a method that does the dynamic-only bit, purely as a convenience wrapper around the `VirtualDispatch` library.

Reply:
done.